src/main/webapp/scripts/mapscript.js

```diff
@@ -0,0 +1,57 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/* ------------ Modal Box */
+
+/* Set Up Map */
+function initMap() {
+  /* 'Hard Coded' origin, Google UK Pancras Square*/
+  const origin = {
+    lat: 51.533364, 
+    lng: -0.125777
+  }
+  const map = new google.maps.Map(document.getElementById("map"), {
+    center: origin,
+    zoom: 15
+  });
+  map.addListener("click", e => {
+    placeMarkerAndPanTo(e.latLng, map);
+  });
+}
+
+/* Place Marker Where Map is Clicked On & Show Popup*/
+function placeMarkerAndPanTo(latLng, map) {
+  var marker = new google.maps.Marker({
+    position: latLng,
+    map: map
+  });
+  map.panTo(latLng);
+  togglePopup();
+  var markerPosition = marker.getPosition();
+  populateLocation(markerPosition);
+
+  /* If marker is right-clicked, delete */
+  google.maps.event.addListener(marker, 'rightclick', function(event) {
+      marker.setMap(null);
+  });
+}
+
+/* Set the PopUp to Active */
+function togglePopup() {
+  document.getElementById("popup-add-recs").classList.toggle("active");
+}
+
+/* Use the position of marker on map to auto-fill location */
+function populateLocation(pos) {
+  document.getElementById("location").value = pos.lat() + ", " + pos.lng()
```

Comment:
should we have two separate elements here for each value? Just to avoid parsing them again on the server

Reply:
sounds good, thanks!